### PR TITLE
Support blockHash filter in eth_getLogs JSON-RPC method

### PR DESCRIPTION
**Pull request by marcciosilva**
## Description

Adds `blockHash` filter option to `eth_getLogs` JSON-RPC method, for convenience.

## Motivation and Context

Fixes issue #1518 

## How Has This Been Tested?

Unit testing.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/rpc/modules/eth/subscribe/EthSubscribeLogsParams.java [hunk 1]
```diff
@@ -41,6 +41,7 @@ public EthSubscribeLogsParams() {
     }
 
     @JsonCreator
+    // TODO: modify?
```
Comment (marcciosilva) [inline rskj-core/src/main/java/co/rsk/rpc/modules/eth/subscribe/EthSubscribeLogsParams.java:44]: @fedejinich should I modify this class as well?
I assume clients can subscribe to events and receive logs somehow, and this is what it's for (or maybe it's deprecated, I really can't tell).

File: rskj-core/src/test/java/org/ethereum/rpc/Web3ImplLogsTest.java [hunk 2]
```diff
@@ -804,13 +806,50 @@ public void getLogsTwiceFromBlockchainWithCallContractAndFilterBySecondTopic() t
...
-        Assert.assertEquals(address, ((LogFilterElement) logs[0]).address);
-        Assert.assertEquals("0x" + INC_EVENT_SIGNATURE, ((LogFilterElement) logs[0]).topics[0]);
-        Assert.assertEquals("0x" + ONE_TOPIC, ((LogFilterElement) logs[0]).topics[1]);
+        assertEquals(address, ((LogFilterElement) logs[0]).address);
+        assertEquals("0x" + INC_EVENT_SIGNATURE, ((LogFilterElement) logs[0]).topics[0]);
+        assertEquals("0x" + ONE_TOPIC, ((LogFilterElement) logs[0]).topics[1]);
+    }
+
+    @Test
+    public void getLogsFromBlockchainWithEventInContractCreationReturnsAsExpectedWithBlockHashFilter() throws Exception {
```
Comment (fedejinich) [inline rskj-core/src/test/java/org/ethereum/rpc/Web3ImplLogsTest.java:819]: nice!

File: rskj-core/src/test/java/org/ethereum/rpc/Web3ImplLogsTest.java [hunk 3]
```diff
@@ -804,13 +806,50 @@ public void getLogsTwiceFromBlockchainWithCallContractAndFilterBySecondTopic() t
...
+
+    @Test(expected = RskJsonRpcRequestException.class)
+    public void getLogsThrowsExceptionWhenBlockHashIsUsedCombinedWithToBlock() throws Exception {
+        addEventInContractCreation();
+        Web3.FilterRequest fr = new Web3.FilterRequest();
+        fr.toBlock = "latest";
+        fr.blockHash = "0xafb368a4f74e51a3c6b6d72b049c4fc7bc7506251f13a3afa4fee4bece0e85eb";
+
+        web3.eth_getLogs(fr);
     }
```
Comment (fedejinich) [inline rskj-core/src/test/java/org/ethereum/rpc/Web3ImplLogsTest.java:854]: should we also cover an unexisting block hash case?

**Review by fedejinich — CHANGES_REQUESTED** [review]

**Comment by marcciosilva** [comment]
pipeline: run

File: rskj-core/src/test/java/org/ethereum/rpc/Web3ImplLogsTest.java (hunk 3 above)
Comment (fedejinich) [reply rskj-core/src/test/java/org/ethereum/rpc/Web3ImplLogsTest.java:871]: :D

**Review by fedejinich — DISMISSED** [review]
lgtm!

**Comment by fedejinich** [comment]
pipeline:run

**Comment by marcciosilva** [comment]
pipeline: run

**Comment by marcciosilva** [comment]
pipeline: run

**Comment by fedejinich** [comment]
pipeline:run

**Review by fedejinich — APPROVED** [review]
lgtm!

**Pull request merged**